Add missing characters to URL regex

daniel-stoneuk opened the pull request:
Added the unreserved character ~ and the reserved characters '(), to the URL matching regex. If I'm understanding [RFC 3986](https://www.rfc-editor.org/rfc/rfc3986#section-2) correctly, "reserved" characters are allowed unencoded in URIs (& there already exist some of these characters in the regex) and so are "unreserved" characters like ~.

The tilde looks like its an unsafe character to have unencoded in RFC 1738, though it appears this is now superseded in most applications - and regardless, if this regex is only used to determine if a link is clickable I don't think there's much harm in including it. I'd love to get some more opinions on this.

This should fix #4705 and improve detection of clickable URLs. I have checked that this is the only place this REGEX string is used but since this is my first contribution, please can someone else check this as well.

Changelog(Fixes): Fixed an issue where URLs in JSON responses were not properly clickable if they included certain unreserved characters

johnwchadwick reviewed (approved):
Looks like it works for me.

Before:
![screenshot showing a hyperlink cut off in Insomnia (develop)](https://user-images.githubusercontent.com/86682572/164491334-c9be3bf3-3a76-4bc3-9558-5e8a99d7dcbd.png)

After:
![screenshot showing a hyperlink that is properly linked in Insomnia (#4710)](https://user-images.githubusercontent.com/86682572/164496799-ebdb0a8b-805f-4c45-8cd7-f3ab5e3bbccf.png)

I don't see a reason why this would break anything either. Obviously, it will break if a URL is followed directly by any of the newly added characters, but that's not a big deal I don't think, especially given that in Insomnia, we're mostly dealing with URLs inside machine-readable formats.

LGTM.

Commit:
Add missing characters to URL regex

*Added the unreserved character ~ and the reserved characters '(), to the URL matching regex since these can appear without being URL encoded.

*Added test for tilde in URL

daniel-stoneuk commented:
Apologies for the force push. I messed up trying to rebase on top of `develop` now that the directory these changes are in has been renamed 😨

johnwchadwick commented:
> Apologies for the force push. I messed up trying to rebase on top of `develop` now that the directory these changes are in has been renamed fearful

Thanks for the rebase. Force pushing on PR branches is fine and not a problem in my opinion.
